### PR TITLE
Added Method to verify Token Online

### PR DESCRIPTION
**Pull request by VarshaKZBlocks**
Response:

jwt expired/ invalid:
`
{
"active":false
}
`
Success: 
`
{
    "exp": 1677495869,
    "iat": 1677494969,
    "jti": "efb9bb87-ac53-4ab8-9ae5-2987d2269bb4",
    "iss": "https://dplatdev.zbyte.io/kc/realms/community",
    "aud": "account",
    "sub": "38b4a564-4c91-4ec3-b548-0637c7782ffb",
    "typ": "Bearer",
    "azp": "lcnc-app",
    "session_state": "a4371e24-e979-48fc-bad1-3bd2f7a44062",
    "name": "varsha kamble",
    "given_name": "varsha",
    "family_name": "kamble",
    "preferred_username": "varsh008@gmail.com",
    "email": "varsh008@gmail.com",
    "email_verified": true,
    "acr": "1",
    "allowed-origins": [
        "https://dplatdev.zbyte.io/",
        "https://dev.zbyte.io/",
        "http://localhost:3000/"
    ],
    "realm_access": {
        "roles": [
            "zbyte_admin",
            "offline_access",
            "app-admin",
            "default-roles-community",
            "developer",
            "uma_authorization"
        ]
    },
    "resource_access": {
        "Account": {
            "roles": [
                "manage-account",
                "manage-account-links",
                "view-profile"
            ]
        }
    },
    "scope": "openid email profile",
    "sid": "a4371e24-e979-48fc-bad1-3bd2f7a44062",
    "client_id": "lcnc-app",
    "username": "varsh008@gmail.com",
    "active": true
}
`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,9 +6,28 @@
...
 2. Use the new access token verification as follows
 ```go
-srv.GET("/a", func(ctx *gin.Context) {
-  access_token := ctx.GetHeader("Authorization")
-	isValid, err := middleware.MiddlewareHandler.VerifyOffline(access_token, "https://dplatdev.zbyte.io/kc")
+srv.GET("/exampleUsingVerifyOffline", func(ctx *gin.Context) {
+  	access_token := ctx.GetHeader("Authorization")
+  	baseUrl:= "https://dplatdev.zbyte.io/kc"
+
+	isValid, err := middleware.MiddlewareHandler.VerifyOffline(access_token,baseUrl)
```
Comment (sourabhmandal) [inline README.md:13]: would be nice, if we take, realm name as a parameter

File: README.md [hunk 2]
```diff
@@ -6,9 +6,28 @@
...
+})
+
+
+srv.GET("/exampleUsingVerifyOnline", func(ctx *gin.Context) {
+  	access_token := ctx.GetHeader("Authorization")
+  	baseUrl:="https://dplatdev.zbyte.io/kc"
+  	client_id:= "lcnc-app"
+  	client_secret:="8T4QEaE5aRIbn9bChDPyVIq4dnjJsxaW"
+
+	isValid, err := middleware.MiddlewareHandler.VerifyOnline(access_token,baseUrl ,clientId, client_secret)
```
Comment (sourabhmandal) [inline README.md:30]: would be nice, if we take, realm name as parameter for verify online too

File: middleware/verify.go [hunk 3]
```diff
@@ -86,3 +90,77 @@ func (m *middlewareStruct) VerifyOffline(accessToken string, baseUrl string) (jw
...
+	//Check if accessToken containe "Bearer" at start, if so remove it to verify
+	if strings.HasPrefix(strings.ToLower(accessToken)[:7], "bearer") {
+		accessToken = accessToken[7:]
+	}
+
+	//Create a http client to make POST request
+	client := &http.Client{}
+
+	//form introspect URL using baseUrl passed
+	keycloak_introspect_url := fmt.Sprintf("%s/realms/community/protocol/openid-connect/token/introspect", baseUrl)
```
Comment (sourabhmandal) [inline middleware/verify.go:130]: shall we save this in a constants file?

**Review by sourabhmandal — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (asbhagatzbyte) [reply README.md:13]: @VarshaZbyte I agree.

File: middleware/structs.go [hunk 4]
```diff
@@ -7,3 +7,35 @@ type VerifyJwtOfflineTokenResponse struct {
 	Header map[string]interface{} `json:"tokenHeader"`
 	Data   jwt.Claims             `json:"data"`
 }
+type VerifyJwtOnlineResponseKeycloak struct {
```
Comment (asbhagatzbyte) [inline middleware/structs.go:10]: Can verify online and verify offline returns data be same?

Comment (VarshaKZBlocks) [reply middleware/structs.go:10]: Both returns data in different format @asbhagatzbyte

File: README.md (hunk 2 above)
Comment (VarshaKZBlocks) [reply README.md:30]: Done

**Review by asbhagatzbyte — APPROVED** [review]

**Pull request merged**